src/DotNetty.Codecs/MessageToMessageDecoder.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace DotNetty.Codecs
+{
+    using System;
+    using System.Collections.Generic;
+    using DotNetty.Common;
+    using DotNetty.Common.Utilities;
+    using DotNetty.Transport.Channels;
+
+    /// <summary>
+    /// Message to message decoder.
+    /// </summary>
+    public abstract class MessageToMessageDecoder<T> : ChannelHandlerAdapter
+    {
+        public bool AcceptInboundMessage(object msg)
```

Comment:
I see AcceptOutboundMessage and there AcceptOutboundMessage is not virtual. It's normal or need to fix too?